[dev-launcher] Add defaultServerUrl option to auto-connect to a baked-in dev server (#46980)
expo/expo: docs/pages/versions/unversioned/sdk/dev-client.mdx

      default: '"most-recent"',
      platform: 'ios',
    },
    {
      name: 'defaultServerUrl',
      description: [

>         'A development server URL (for example `http://192.168.1.50:8081`) to bake into the build and auto-connect to on launch, instead of waiting for the user to scan a QR code or pick a development server. Useful for headless, CI, or agent-driven workflows, or when running several Metro servers on fixed, dedicated ports that are known ahead of build time.',

Reviewer: **[suggestion]** This sentence runs about 33 words and packs three ideas (what the URL is, what it does, and what it replaces). One thought per sentence reads better mid-task, so split it (`style-guide.md#voice-and-tone`).

```suggestion
        'A development server URL (for example `http://192.168.1.50:8081`) to bake into the build and auto-connect to on launch. The launcher uses it instead of waiting for the user to scan a QR code or pick a development server. Useful for headless, CI, or agent-driven workflows, or when running several Metro servers on fixed, dedicated ports that are known ahead of build time.',
```

>         'If the server is unreachable, the launcher falls back to its home screen, and a recently opened project still takes precedence (unless `launchMode` is `launcher`). Can be set per-platform with `ios.defaultServerUrl` / `android.defaultServerUrl`, or overridden at build time with the `EXPO_DEV_LAUNCHER_DEFAULT_SERVER_URL` environment variable.',

Reviewer: **[suggestion]** Three small fixes on this line. (1) Prefer active voice over the passive "Can be set ... or overridden" (`style-guide.md#voice-and-tone`); the surrounding entries use the active "Set this property to ..." form. (2) Use Android-before-iOS platform order (`style-guide.md#platform-order`); the existing table already lists `android.launchMode` before `ios.launchMode`. (3) Drop the spaces around the slash (`style-guide.md#punctuation`).

```suggestion
        'If the server is unreachable, the launcher falls back to its home screen, and a recently opened project still takes precedence (unless `launchMode` is `launcher`). Set it per-platform with `android.defaultServerUrl`/`ios.defaultServerUrl`, or override it at build time with the `EXPO_DEV_LAUNCHER_DEFAULT_SERVER_URL` environment variable.',
```
